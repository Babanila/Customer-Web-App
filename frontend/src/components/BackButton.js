import React from 'react'
import { css } from 'emotion'
import SingleButton from './SingleButton'

function BackButton({ handleBack }) {
  const backBtnStyle = css`
    padding: 10px;
    font-size: 1em;
    color: #000000;
    background-color: #ffffff;
    border: 2px solid #000000;
    border-radius: 5px;
    outline: none;
    &:hover {
      cursor: pointer;
      color: #ffffff;
      background-color: #000000;
    }
  `
  return <SingleButton btnName="Back" btnClick={handleBack} btnStyles={backBtnStyle} />
}

export default BackButton
