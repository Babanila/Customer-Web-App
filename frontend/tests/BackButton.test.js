import React from 'react'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import { mount } from 'enzyme'
import BackButton from '../src/components/BackButton'

describe('Check BackButton Component', () => {
  const handleBack = jest.fn()

  it('should renders without crashing', () => {
    const tree = renderer.create(<BackButton handleBack={handleBack} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correct number of element in BackButton component', () => {
    const BackButtonWrapper = mount(<BackButton handleBack={handleBack} />)
    const actual = BackButtonWrapper.find('SingleButton')

    expect(BackButton).toBeDefined()
    expect(actual.exists).toBeTruthy()
    expect(actual.length).toEqual(1)
    expect(actual.find('button').length).toEqual(1)
    expect(actual.find('button').text()).toEqual('Back')
  })

  it('should renders correct text in BackButton component', () => {
    const { getByText } = render(<BackButton handleBack={handleBack} />)
    expect(getByText(/back/i)).toHaveTextContent('Back')
  })
})
