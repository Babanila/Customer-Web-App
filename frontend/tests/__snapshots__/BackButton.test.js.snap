// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check BackButton Component should renders without crashing 1`] = `
<button
  className="css-14mojuj"
  onClick={[Function]}
>
  Back
</button>
`;
